openlibrary/core/vendors.py

```diff
@@ -454,30 +450,31 @@ def cached_get_amazon_metadata(*args, **kwargs):
 
 
 @public
-def get_betterworldbooks_metadata(isbn: str) -> Optional[dict]:
+def get_betterworldbooks_metadata(input_isbn: str) -> dict | None:
     """
-    :param str isbn: Unnormalisied ISBN10 or ISBN13
+    :param str input_isbn: Unnormalisied ISBN10 or ISBN13
     :return: Metadata for a single BWB book, currently listed on their catalog, or
              an error dict.
-    :rtype: dict or None
     """
 
-    isbn = normalize_isbn(isbn)
+    isbn = normalize_isbn(input_isbn)
```

Comment:
Why two different variables?  I don't see where this extra code buys us anything.

Reply:
I realize I am probably not handling this properly, but that was my attempt to address the `isbn` parameter to `get_betterworldbooks_metadata(isbn: str)` being a string, and the output from `normalize_isbn()` being `str | None`. When the ISBN is bad on its face, even prior to the checksum, `normalize_isbn()` will return `None`.

Without this change I was getting:
```
openlibrary/core/vendors.py: note: In function "get_betterworldbooks_metadata":
openlibrary/core/vendors.py:460: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")  [assignment]
        isbn = normalize_isbn(isbn)
               ^
```

My other attempt was as follows:
```
    possible_isbn = normalize_isbn(isbn)
    if possible_isbn is None:
        return None
    isbn = possible_isbn
```


For context, here's `normalize_isbn()`.
```
def normalize_isbn(isbn: str) -> str | None:
    """
    Takes an isbn-like string, keeps only numbers and X/x, and returns an ISBN-like
    string or None.
    Does NOT validate length or checkdigits.
    """
    return isbn and canonical(isbn) or None
```

Reply:
Ah, what a good idea. Perfect. Thank you for suggesting that. I will make these changes. :)